tutorials/02-Basic Operations.ipynb

```diff
@@ -0,0 +1,1822 @@
+    "\n",
+    "(with automatic evaluation)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 55,
+   "id": "eaadbd08-8178-47fb-a387-71913a3b6d8a",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/latex": [
+       "$\\displaystyle y + \\left(y + 1\\right)^{2} + 2$"
+      ],
+      "text/plain": [
+       "y + (y + 1)**2 + 2"
+      ]
+     },
+     "execution_count": 55,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "x = Symbol('x')\n",
+    "y = Symbol('y')\n",
+    "expr = x**2 + x + 1\n",
+    "expr.subs(x, y + 1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 56,
+   "id": "dbc61768-f0fd-4133-8665-9c49dab8d6e3",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/latex": [
+       "$\\displaystyle f^{2}{\\left(y + 1 \\right)} + f{\\left(y + 1 \\right)} + 1$"
+      ],
+      "text/plain": [
+       "f(y + 1)**2 + f(y + 1) + 1"
+      ]
+     },
+     "execution_count": 56,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "f = Function('f')\n",
+    "\n",
+    "expr.subs(x, f(y + 1))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "70aa16f9-b153-4356-9565-647997c7310d",
+   "metadata": {},
+   "source": [
+    "Multivariate substitution also works\n",
+    "(with dictionary)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 57,
+   "id": "084ec64a-19ed-4596-97a7-54aa876de53e",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/latex": [
+       "$\\displaystyle f{\\left(f{\\left(1 \\right)},g{\\left(f{\\left(1 \\right)},2 \\right)} \\right)}$"
+      ],
+      "text/plain": [
+       "f(f(1), g(f(1), 2))"
+      ]
+     },
+     "execution_count": 57,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "expr = f(x, y)\n",
+    "g = Function('g')\n",
+    "\n",
+    "expr.subs({x: f(1), y: g(f(1), 2)})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ba8cbb60-8338-41e3-9353-95dfe72ac91d",
+   "metadata": {},
+   "source": [
+    "The source of substitution can be more general than symbols"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 61,
+   "id": "cc5516e2-6c86-4950-af1f-9abac305c865",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/latex": [
+       "$\\displaystyle g{\\left(y \\right)} + h{\\left(z \\right)} + 1$"
+      ],
+      "text/plain": [
+       "g(y) + h(z) + 1"
+      ]
+     },
+     "execution_count": 61,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "expr = f(x + 1) + g(y) + 1\n",
+    "h = Function('h')\n",
+    "z = Symbol('z')\n",
+    "\n",
+    "expr.subs(f(x + 1), h(z))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9c8ca052-b431-46ff-a41b-2f4badf31979",
+   "metadata": {},
+   "source": [
+    "The formal definition of sympy `subs` is:\n",
```

Comment:
This is confusing, and I don't think it really adds anything to the discussion. I would just delete it.